client/scripts/app.ts

```diff
@@ -137,28 +137,36 @@ export function handleUpdateEmailConfirmation() {
   }
 }
 
-export async function selectCommunity(c?: CommunityInfo): Promise<void> {
+export async function selectCommunity(c?: CommunityInfo): Promise<boolean> {
```

Comment:
what is being returned? should explain in the comment

Reply:
This boolean return is largely non-functional, but was added to mirror the selectNode function (where the return is necessary to prevent further chain initialization fns being called)—my instinct was that the symmetry added predictability/reduced eccentricity in the codebase. Thoughts on keeping/changing/documenting?

Reply:
I don't follow. If no value is being returned, then we shouldn't return a boolean for the sake of symmetry. I can't tell if your design decision here is being made with an awareness of why selectNode is returning a boolean (it really should be but you haven't mentioned it). Can we try going through an explanation again while noting that?

Reply:
A value is being returned; as in selectNode, that value reflects whether the community initialization completed or not. It's just that the return value is never used. It can be accessed if we want to add wrapping logic but I did not yet see a need or purpose.

Reply:
We should probably remove the return value then but I want to understand why the return value isn't used. For this PR we should probably leave out the change to selectCommunity though.

Reply:
All this function needs to do is break, rather than continue, if the route param does not match the id of the community being initialized. It could alternately return void or throw an error from a try/catch structure.

However, selectNode _does_ need to return a boolean, because the initChain fn is called after selectNode to continue chain initialization. That initChain fn should only be called if selectNode completes successfully, i.e. returns true. I opted to have selectCommunity return a boolean, reflecting whether it completes initialization or not, in order to mirror the structure of its chain counterpart, selectNode.

Reply:
To "finalizeInitialization" is to reset the forum stores—clear and then populate them with the newly fetched bulkOffchain data. If the route param does not match upon completion of the bulkOffchain fetch, the stores are not reset.